tiny_dnn/core/kernels/conv2d_grad_op.h

```diff
@@ -22,28 +22,36 @@ class Conv2dGradOp : public core::OpKernel {
   void compute(core::OpKernelContext &context) override {
     auto params = OpKernel::params_->conv();
 
-    // incoming/outcoming data
-    const tensor_t &prev_out = context.input(0);
-    const tensor_t &W        = context.input(1);
-    tensor_t &dW             = context.input_grad(1);
-    tensor_t &db             = context.input_grad(2);
-    tensor_t &prev_delta     = context.input_grad(0);
-    tensor_t &curr_delta     = context.output_grad(0);
+    // TODO(Randl): Remove once layers forward and backward by themself.
+    const Tensor<float_t> prev_out_t(context.input(0)),
+      weights_t(context.input(1));
+    Tensor<float_t> weights_grads_t(context.input_grad(1));
+    Tensor<float_t> bias_grads_t(context.input_grad(2));
+    Tensor<float_t> prev_delta_t(context.input_grad(0));
+    Tensor<float_t> curr_delta_t(context.output_grad(0));
 
-    // initalize outputs
-    fill_tensor(prev_delta, float_t{0});
+    // initialize outputs
+    prev_delta_t.fill(float_t{0});
```

Comment:
use same convention for `float_t{0}` as in https://github.com/tiny-dnn/tiny-dnn/pull/759/files#diff-fc64083110af88933c8ad786595bf36dR33